### PR TITLE
Game Manager, Camera Position y Persistencia de datos

### PR DESCRIPTION
**Pull request by OniriaGames**

### CODE REVIEW

File: Assets/Scripts/GameManager/GameManager.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+		private PlayerInputManager _playerInputManager;
+
+		private void Awake()
+		{
+			Instance = this;
+		}
+
+		private void Start()
+		{
+			_playerInputManager = GetComponent<PlayerInputManager>();
```
Comment (EndoAlejandro) [inline Assets/Scripts/GameManager/GameManager.cs:25]: Try to do get component only in awake

File: Assets/Scripts/GameManager/PlayerCameraSettings.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+		public static PlayerCameraSettings Instance;
+
+		[SerializeField] CinemachineVirtualCamera[] cameras;
+		[SerializeField] GameObject target;
+
+		private int _currentCameraIndex = 0;
+
+		private void Awake()
+		{
+			Instance = this;
```
Comment (EndoAlejandro) [inline Assets/Scripts/GameManager/PlayerCameraSettings.cs:18]: Check instance or singleton

File: Assets/Scripts/GameManager/PlayerCarSelection.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace InGame {
+	public class PlayerCarSelection : MonoBehaviour
+	{
+		public static PlayerCarSelection Instance;
+		public static int playerID = 0;
+		[SerializeField] CarAssets carAssets;
```
Comment (EndoAlejandro) [inline Assets/Scripts/GameManager/PlayerCarSelection.cs:11]: always add moddifier (private)

File: Assets/Scripts/Menu/MenuManager.cs [hunk 4]
```diff
@@ -24,22 +34,44 @@ public class MenuManager : MonoBehaviour
 		[SerializeField] Slider sliderGlobalVolume;
 		[Space(5)]
 		[Header("Language")]
-		[SerializeField] TextMeshProUGUI _languageText;
+		[SerializeField] TextMeshProUGUI languageText;
```
Comment (EndoAlejandro) [inline Assets/Scripts/Menu/MenuManager.cs:37]: Add privae

File: Assets/Scripts/Menu/MenuManager.cs [hunk 5]
```diff
@@ -24,22 +34,44 @@ public class MenuManager : MonoBehaviour
...
 		//LANGUAGE
 		private int _currentLocaleIndex;
 
+		//CUPS
+		private int _selectedCupIndex;
+
+		//Player ready
+		private int _playersReady;
+
+		public int PlayersReady {
```
Comment (EndoAlejandro) [inline Assets/Scripts/Menu/MenuManager.cs:56]: Try change for a method

File: Assets/Scripts/Menu/MenuManager.cs [hunk 6]
```diff
@@ -24,22 +34,44 @@ public class MenuManager : MonoBehaviour
...
+			}
+		}
+
+		public int SelectedCupIndex { set { _selectedCupIndex = value; } }
+
+
 		private void Awake()
 		{
-			audioSource = GameObject.FindGameObjectWithTag("SoundFX").GetComponent<AudioSource>();
+			Instance= this;
```
Comment (EndoAlejandro) [inline Assets/Scripts/Menu/MenuManager.cs:72]: Singleton

File: Assets/Scripts/Menu/MenuManager.cs [hunk 7]
```diff
@@ -89,31 +121,47 @@ IEnumerator Start()
...
+			int index = 0;
+
+			for (int i = 0; i < allCupsAssets.cups.Length;i++) {
+				if (allCupsAssets.cups[i].cupID == PlayerPrefs.GetInt("CurrentCupID")) { 
+					index= i;
+					break;
+				}
+			}
+
+			PlayerPrefs.SetInt("CurrentSpeedway",0);
```
Comment (EndoAlejandro) [inline Assets/Scripts/Menu/MenuManager.cs:134]: Try to centralize player pref operations

File: Assets/Scripts/Menu/Player.cs [hunk 8]
```diff
@@ -2,20 +2,23 @@
...
 namespace Menu {
 	public class Player : MonoBehaviour
 	{
 		public static int playerID = 0;
-		public TextMeshProUGUI _playerIDText;
-		public Transform _carTransform;
-		public CarAssets carAssets;
+
+		[SerializeField] TextMeshProUGUI playerIDText;
+		[SerializeField] Transform carTransform;
```
Comment (EndoAlejandro) [inline Assets/Scripts/Menu/Player.cs:13]: Add private

File: Assets/Scripts/GameManager/GameManager.cs [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+	public class GameManager : MonoBehaviour
+	{
+		public static GameManager Instance;
+		[SerializeField] private AllCupsAssets allCupsAssets;
+
+		private PlayerInputManager _playerInputManager;
+
+		private void Awake()
+		{
+			Instance = this;
```
[the remark this answers is not in the document]
Comment (OniriaGames) [reply Assets/Scripts/GameManager/GameManager.cs:20]: Corregido

File: Assets/Scripts/GameManager/GameManager.cs (hunk 1 above)
Comment (OniriaGames) [reply Assets/Scripts/GameManager/GameManager.cs:25]: Corregido

File: Assets/Scripts/GameManager/PlayerCameraSettings.cs (hunk 2 above)
Comment (OniriaGames) [reply Assets/Scripts/GameManager/PlayerCameraSettings.cs:18]: Corregido

File: Assets/Scripts/GameManager/PlayerCarSelection.cs (hunk 3 above)
Comment (OniriaGames) [reply Assets/Scripts/GameManager/PlayerCarSelection.cs:11]: Corregido

File: Assets/Scripts/Menu/MenuManager.cs (hunk 4 above)
Comment (OniriaGames) [reply Assets/Scripts/Menu/MenuManager.cs:37]: Corregido

File: Assets/Scripts/Menu/MenuManager.cs (hunk 5 above)
Comment (OniriaGames) [reply Assets/Scripts/Menu/MenuManager.cs:56]: Corregido

File: Assets/Scripts/Menu/MenuManager.cs (hunk 6 above)
Comment (OniriaGames) [reply Assets/Scripts/Menu/MenuManager.cs:72]: Corregido

File: Assets/Scripts/Menu/MenuManager.cs (hunk 7 above)
Comment (OniriaGames) [reply Assets/Scripts/Menu/MenuManager.cs:134]: Corregido

File: Assets/Scripts/Menu/Player.cs (hunk 8 above)
Comment (OniriaGames) [reply Assets/Scripts/Menu/Player.cs:13]: Corregido

**Pull request merged**